app/components/Upload/ImageUpload.js

```diff
@@ -46,27 +46,40 @@ type UploadAreaProps = {
   image: ?string,
 };
 
-const FilePreview = ({ file, onRemove, index }: FilePreviewProps) => (
-  <Flex
-    wrap
-    className={styles.previewRow}
-    alignItems="center"
-    justifyContent="space-between"
-  >
-    <div style={{ width: '90%' }}>
-      <TextInput
-        disabled
-        value={file.name}
-        style={{ width: '100%', height: 50 }}
+const FilePreview = ({ file, onRemove, index }: FilePreviewProps) => {
+  const [previewUrl, setPreviewUrl] = useState();
+  useEffect(() => {
+    !previewUrl && setPreviewUrl(URL.createObjectURL(file));
+    return () => {
+      if (previewUrl) URL.revokeObjectURL(previewUrl);
+    };
+  }, [file, previewUrl]);
+
+  return (
+    <Flex
+      wrap
+      className={styles.previewRow}
+      alignItems="center"
+      justifyContent="space-between"
+    >
+      <div style={{ width: '90%', display: 'flex' }}>
+        <img width="60" alt="preview" src={previewUrl} />
+        <div style={{ flex: '1 1 auto' }}>
+          <TextInput
```

Comment:
I find it a bit weird to use the `<TextInput />` - just print out the file name like a normal kid

Reply:
I think it's just to get the style of the input field. Having it disabled just means we can reuse the styling without any more work. Kinda weird, but also kinda nice.

Reply:
Ye i get the styling thing, but it's just weird and bad practice

Reply:
I disagree. It's a pattern we use all over the app, and I think it makes sense for items that are "user-defined" but not possible to change directly. (the same is done for emails, f.ex.

But I see your point in it being kinda weird. Out of scope for this PR though.